edx_event_bus_kafka/internal/consumer.py

```diff
@@ -303,25 +308,18 @@ def _consume_indefinitely(self):
         finally:
             self.consumer.close()
 
-    def consume_indefinitely(self, offset_timestamp=None):
+    def consume_indefinitely(self):
         """
         Consume events from a topic in an infinite loop.
-
-        Arguments:
-            offset_timestamp (datetime): Optional and deprecated; if supplied, calls
-                ``reset_offsets_and_sleep_indefinitely`` instead. Relying code should
-                switch to calling that method directly.
         """
-        # TODO: Once this deprecated argument can be removed, just
-        # remove this delegation method entirely and rename
-        # `_consume_indefinitely` to no longer have the `_` prefix.
-        if offset_timestamp is None:
+        if self.offset_time is None:
             self._consume_indefinitely()
         else:
-            warnings.warn(
-                "Calling consume_indefinitely with offset_timestamp is deprecated; "
-                "please call reset_offsets_and_sleep_indefinitely directly instead."
-            )
+            try:
+                offset_timestamp = datetime.fromisoformat(str(self.offset_time))
+            except ValueError:
+                logger.exception('Could not parse the offset timestamp.')
+                raise
             self.reset_offsets_and_sleep_indefinitely(offset_timestamp)
```

Comment:
I think it would be best to move the parsing code to the constructor (where the value is first introduced) so that the parsed timestamp can be stored on the instance. Then we can drop the redundant passing of `offset_timestamp` to the method and have it just pull from `self` (and be directly callable on its own).

Reply:
Updated.